bot/cogs/defcon.py

```diff
@@ -107,6 +115,18 @@ def mod_log(self) -> ModLog:
         """Check the DEFCON status or run a subcommand."""
         await ctx.invoke(self.bot.get_command("help"), "defcon")
 
+    async def _defcon_action(self, ctx: Context, action: Awaitable[None], action_type: str) -> None:
```

Comment:
Given that the action is always a PUT request, this may as well just be done here and take parameters for the days and enable/disable. 

Perhaps then you could also get rid of `action_type` and deduce it from the aforementioned arguments. Going with the assumption that the days would never be updated to 0, could "updated" be when enabled = `True` and days = `0`? An alternative is to keep `action_type` and not have an `enabled` parameter since I think it's easy to deduce that value from the value of `action_type`.

Reply:
That's some very good suggestions, now that I inspected the actions closer, it is very possible to move all the puts there, and to accept only days and the bool for `'enabled''. Thank you.